### PR TITLE
chore: add typecheck and workspace examples

### PR DESCRIPTION
**Pull request by hi-ogawa**
### Description

I'm not sure if these examples are worth inclusion in the official list, but I thought having this somewhere would be convenient to quickly spawn reproduction on stackblitz for issue authors and also for maintainers.

- https://stackblitz.com/github/hi-ogawa/vitest/tree/chore-repro-templates/examples/workspace
- https://stackblitz.com/github/hi-ogawa/vitest/tree/chore-repro-templates/examples/typecheck

_todo_ 

- [x] docs
- [x] ~readme~ `pnpm docs:examples`

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [ ] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

**Review by AriPerkkio — COMMENTED** [review]
There's quite many changes in lockfile. 🤔 
I've usually just manually picked the needed changes from there are staged them line-by-line.

Otherwise this looks good to me. It would be great if we had these runnable examples (Stackblitz links) in configuration documentation directly.

**Comment by hi-ogawa** [comment]
> There's quite many changes in lockfile. 🤔 I've usually just manually picked the needed changes from there are staged them line-by-line.

Ah, I didn't know that. I've been just pushing everything after `pnpm i`. Also now looking at the lock file, I might be installing different version of `@types/node`. Let me check that as well.

> It would be great if we had these runnable examples (Stackblitz links) in configuration documentation directly.

Yeah, that would be nice. Since there is already a full list here https://vitest.dev/guide/#examples, so maybe putting internal links from guide pages (https://vitest.dev/guide/testing-types.html and https://vitest.dev/guide/workspace.html) to https://vitest.dev/guide/#examples is enough?

File: examples/README.md [hunk 1]
```diff
@@ -22,3 +22,5 @@
 | `vitesse` | [GitHub](https://github.com/vitest-dev/vitest/tree/main/examples/vitesse) | [Play Online](https://stackblitz.com/fork/github/vitest-dev/vitest/tree/main/examples/vitesse?initialPath=__vitest__/) |
 | `vue-jsx` | [GitHub](https://github.com/vitest-dev/vitest/tree/main/examples/vue-jsx) | [Play Online](https://stackblitz.com/fork/github/vitest-dev/vitest/tree/main/examples/vue-jsx?initialPath=__vitest__/) |
 | `vue` | [GitHub](https://github.com/vitest-dev/vitest/tree/main/examples/vue) | [Play Online](https://stackblitz.com/fork/github/vitest-dev/vitest/tree/main/examples/vue?initialPath=__vitest__/) |
+| `typecheck` | [GitHub](https://github.com/vitest-dev/vitest/tree/main/examples/typecheck) | [Play Online](https://stackblitz.com/fork/github/vitest-dev/vitest/tree/main/examples/typecheck?initialPath=__vitest__/) |
```
Comment (sheremet-va) [inline examples/README.md:25]: Did you use a generator for that from scripts folder? I think generator sorts it by name

**Review by sheremet-va — APPROVED** [review]
Good idea 👍

File: docs/guide/testing-types.md [hunk 2]
```diff
@@ -4,6 +4,18 @@ title: Testing Types | Guide
...
+[GitHub](https://github.com/vitest-dev/vitest/tree/main/examples/typecheck) - [Play Online](https://stackblitz.com/fork/github/vitest-dev/vitest/tree/main/examples/typecheck?initialPath=__vitest__/)
+
+:::
+
+::: tip
+
+You can find a sample project in [Examples](/guide/#examples).
+
+:::
+
```
Comment (hi-ogawa) [inline docs/guide/testing-types.md:18]: I tested these two patterns and probably having direct links looks nicer. What do you think?

image

image

Comment (sheremet-va) [reply docs/guide/testing-types.md:18]: I like the first pattern

File: examples/README.md (hunk 1 above)
Comment (hi-ogawa) [reply examples/README.md]: Thanks. I didn't know there is `pnpm docs:examples` script. It looks like the last "marko" one was also added manually https://github.com/vitest-dev/vitest/pull/5006/commits/e75dabb5e57ba20dffc708f3faf8de21c46f3845

**Review by sheremet-va — APPROVED** [review]

**Pull request merged**